### PR TITLE
job-manager: do not checkpoint on every queue state change

### PR DESCRIPTION
**Pull request by chu11**
Per #4804, the number of checkpoints to the KVS seems a bit excessive, as it occurs on every change.

Instead, do it only when the user asks to via a `sync` flag.  So that users can say "hey preserve this specific change just in case of catastrophic failure", as well as to help with testing.

I made one additional change that I think is important, when checkpointing in the `job-manager`, make sure that the FLUX_KVS_SYNC flag is set, to ensure the data is on the backing store as well.  Would have slowed things down before, but now that syncs are optional, I think it more than offsets it.

### CODE REVIEW

File: src/modules/job-manager/restart.c
Comment (garlick) [inline src/modules/job-manager/restart.c:244]: Again - high cost operation, and we already have `kvs.checkpoint-period` periodically saving the KVS root.

File: src/cmd/flux-queue.c
Comment (garlick) [inline src/cmd/flux-queue.c:54]: As noted previously I think this option is not something an admin needs/wants.

**Review by garlick — CHANGES_REQUESTED** [review]
As noted inline, I think the KVS_SYNC flag is inappropriate, and the user-facing option is probably not the right way to go.   However, maybe the change suggested in #4856 is going too far in the opposite direction?

If we left the KVS commit of queue state changes as is, then at least `kvs.checkpoint-period` would make sure the backing store is kept reasonably current in the event of a crash.  Thoughts?

File: src/modules/job-manager/queue.c
Comment (chu11) [reply src/modules/job-manager/queue.c:618]: Yeah, I basically went down this path solely to keep some current tests working (they check `checkpoint.job-manager` directly after a `flux queue enable/disable/start/stop`.  Will re-do and just re-work the tests instead!

**Comment by chu11** [comment]
re-pushed, going with an alternate approach.  Simply do not checkpoint on every queue state change, only checkpoint when the `job-manager` module is unloaded.

Had to update some tests as a result.

**Review by garlick — APPROVED** [review]
LGTM!

**Comment by garlick** [comment]
Any reason not to set MWP on this?

**Comment by chu11** [comment]
nope, was just waiting for your approval :-)

**Pull request merged**
